src/main/java/picard/sam/CreateSequenceDictionary.java

```diff
@@ -185,40 +177,15 @@ protected int doWork() {
                     " already exists.  Delete this file and try again, or specify a different output file.");
         }
 
-        // SortingCollection is used to check uniqueness of sequence names
-        final SortingCollection<String> sequenceNames = makeSortingCollection();
         try (BufferedWriter writer = makeWriter()) {
-            final ReferenceSequenceFile refSeqFile = ReferenceSequenceFileFactory.
-                    getReferenceSequenceFile(REFERENCE_SEQUENCE, TRUNCATE_NAMES_AT_WHITESPACE);
-            SAMSequenceDictionaryCodec samDictCodec = new SAMSequenceDictionaryCodec(writer);
-
-            samDictCodec.encodeHeaderLine(false);
-            // read reference sequence one by one and write its metadata
-            for (ReferenceSequence refSeq = refSeqFile.nextSequence(); refSeq != null; refSeq = refSeqFile.nextSequence()) {
-                final SAMSequenceRecord samSequenceRecord = makeSequenceRecord(refSeq);
-                samDictCodec.encodeSequenceRecord(samSequenceRecord);
-                sequenceNames.add(refSeq.getName());
-            }
+            SequenceDictionaryUtils.encodeDictionary(writer, new SAMSequenceRecordIterator(REFERENCE_SEQUENCE, TRUNCATE_NAMES_AT_WHITESPACE));
         } catch (FileNotFoundException e) {
             throw new PicardException("File " + OUTPUT.getAbsolutePath() + " not found");
         } catch (IOException e) {
             throw new PicardException("Can't write to or close output file " + OUTPUT.getAbsolutePath());
-        }
-
-        // check uniqueness of sequences names
-        final CloseableIterator<String> iterator = sequenceNames.iterator();
-
-        if(!iterator.hasNext()) return 0;
-
-        String current = iterator.next();
-        while (iterator.hasNext()) {
-            final String next = iterator.next();
-            if (current.equals(next)) {
-                OUTPUT.delete();
-                throw new PicardException("Sequence name " + current +
-                        " appears more than once in reference file");
-            }
-            current = next;
+        } catch (IllegalArgumentException e) {
+            OUTPUT.delete();
+            throw new PicardException(e.getMessage());
```

Comment:
Hard to tell what's intended here; do you need to export the `IllegalArgumentException` as a `PicardException` instead? Otherwise the `OUTPUT.delete()` code could go in a finally.

Reply:
not good in a finally...since it only needs to happen if something is broken. added a comment to that effect.